pipelines/validate_run.py

```
@@ -89,52 +122,133 @@ def reject_json_constant(value):
 
 
-def check_spike_order(events, where):
-    """Require finite timestamps and global non-decreasing event order."""
+# Marker substring of the inversion message built in check_spike_order.
+# check_records imports it to drop the shape layer's order errors: that layer
+# owns spike order and would otherwise report the same inversion twice.
+SPIKE_ORDER_MISMATCH = "spike_events not globally non-decreasing"
+SPIKE_TIME_KEY_MISMATCH = "spike_events must use one timestamp key throughout"
+
+
+def check_spike_order(events, where, require_keys=BRIDGE_SPIKE_EVENT_KEYS):
```

Ruff (0.16.2)

[warning] Too many branches (14 > 12)

(PLR0912)